nexis/lib/widgets/auth/custom_input_field.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+
+class CustomTextField extends StatelessWidget {
+  final bool? obscureText;
+  final String? hint;
+  final void Function()? onTap;
+  final TextEditingController? controller;
+  final bool? readOnly;
+  final void Function(String)? onSubmitted;
+
+  const CustomTextField({
+    this.obscureText, 
+    this.hint, 
+    this.onTap, 
+    this.controller, 
+    this.readOnly, 
+    this.onSubmitted, 
+    super.key
+    });
```

Comment:
This is not how you implement super.key in the constructor, you would do it like this:
`
const CustomTextField({
    Key? key,
    this.obscureText, 
    this.hint, 
    this.onTap, 
    this.controller, 
    this.readOnly, 
    this.onSubmitted, 
    }) : super(key: key);
`